File: livingrimoire start here/LivinGrimoire java/src/Auxiliary_Modules/TrgTime.java
package Auxiliary_Modules;

public class TrgTime {
    String t = "null";
    private Boolean alarm = true;
    public void setTime(String v1){
        t = RegexUtil.extractRegex(enumRegexGrimoire.simpleTimeStamp,v1);
    }
    public Boolean alarm(){
        String now = TimeUtils.getCurrentTimeStamp();
        if(alarm){
            if(now.equals(t)){
                alarm = false;
                return true;
            }
        }
        if(!now.equals(t)){
            alarm = true;
        }
        return false;
    }
}

File: livingrimoire start here/LivinGrimoire Kotlin/kotlin/auxiliary_modules/TrgTime.kt
package auxiliary_modules

class TrgTime {
    private var t = "null"
    private var alarm = true
    fun setTime(v1: String) {
        t = RegexUtil.extractRegex(enumRegexGrimoire.SimpleTimeStamp, v1)
    }

    fun alarm(): Boolean {
        val now: String = TimeUtils.currentTimeStamp
        if (alarm) {
            if (now == t) {
                alarm = false
                return true
            }
        }
        if (now != t) {
            alarm = true
        }
        return false
    }
}